packages/phone-number-privacy/signer/src/common/identity.ts

```diff
@@ -1,21 +1,46 @@
 import { verifySignature } from '@celo/utils/lib/signatureUtils'
 import { Request } from 'express'
+import { getDataEncryptionKey } from '../web3/contracts'
 import logger from './logger'
 
 /*
  * Confirms that user is who they say they are and throws error on failure to confirm.
  * Authorization header should contain the EC signed body
  */
-export function authenticateUser(request: Request): boolean {
+export async function authenticateUser(request: Request): Promise<boolean> {
   logger.debug('Authenticating user')
 
   // https://tools.ietf.org/html/rfc7235#section-4.2
   const messageSignature = request.get('Authorization')
+  const message = JSON.stringify(request.body)
+  if (!messageSignature || !message) {
+    return false
+  }
   const signer = request.body.account
   if (!messageSignature || !signer) {
     return false
   }
 
-  const message = JSON.stringify(request.body)
+  const registeredEncryptionKey = await getDataEncryptionKey(signer)
+  if (!registeredEncryptionKey) {
+    logger.info(`Account ${signer} does not have registered encryption key`)
+  } else {
+    const validSignature: boolean = verifySignature(
+      message,
+      messageSignature,
+      registeredEncryptionKey
+    )
+    if (validSignature) {
+      return true
```

Comment:
return validSignature?

Reply:
If it's false, I don't want to return it since I want to be able to fallback to the wallet approach. I don't expect Valora to fall into this case, but 3rd parties might call only using the wallet key as an example.